scripts/globals/pirates.lua

```diff
@@ -0,0 +1,267 @@
+-----------------------------------
+-- Pirates helpers
+-----------------------------------
+require("scripts/globals/pathfind")
+
+xi = xi or {}
+xi.pirates = xi.pirates or {}
+xi.pirates.status =
+{
+    IDLE       = 0,
+    PREPARING  = 1,
+    ARRIVING   = 2,
+    SPAWNING   = 3,
+    ATTACKING  = 4,
+    DESPAWNING = 5,
+    DEPARTING  = 6,
+    FINISHED   = 7
+}
+
+local function SetZoneMusic(zone, block, track)
+    for _, player in pairs(zone:getPlayers()) do
+        player:ChangeMusic(block, track)
+    end
+end
+
+xi.pirates.init = function(ID)
+    local ship = GetNPCByID(ID.npc.PIRATE_SHIP.id)
+    ship:setAnimPath(ID.npc.PIRATE_SHIP.anim_path)
+    ship:setStatus(xi.status.DISAPPEAR)
+
+    for k, pirate in pairs(ID.npc.PIRATES) do
+        local npc = GetNPCByID(k)
+        npc:setPos(pirate.start_pos)
+        npc:initNpcAi()
+    end
+end
+
+xi.pirates.start = function(ID)
+    local ship = GetNPCByID(ID.npc.PIRATE_SHIP.id)
+    xi.pirates.setStatus(ship, xi.pirates.status.IDLE)
+    xi.pirates.despawnMobs(ID)
+    xi.pirates.despawnNPCs(ID)
+    xi.pirates.reset(ID)
+end
+
+xi.pirates.getStatus = function(ship)
+    return ship:getLocalVar("pirateStatus")
+end
+
+xi.pirates.setStatus = function(ship, status)
+    ship:setLocalVar("pirateStatus", status)
```

Comment:
This can be done by just doing ship:setLocalVar("PriateStatus") elsewhere, just make a global var up top, then call it. Unlike C++ we don't need a return function because we are not dealing with public, friend, and private classes. 
So you can do something like this in the global:
`xi.pirates.statusvar = "PirateStatus"`

Then in replace the function with:
`ship:setLocalVar(xi.pirates.status, statusvar)`